apps/authentication-service/src/resources/clients/clientsRoutesHandlers.ts

```diff
@@ -0,0 +1,85 @@
+import { Client } from "auth0";
+import { Request, Response } from "express";
+
+import {
+  createClient,
+  deleteClient,
+  getClient,
+} from "@/resources/clients/clientsService";
+import { getClientTokenFromRequest } from "@/shared/utils";
+
+export const clientsCreateV1 = async (req: Request, res: Response) => {
+  const clientToken = getClientTokenFromRequest(req);
+
+  const clientBody = {
+    name: `ucp-${crypto.randomUUID()}`,
+    client_metadata: {
+      created_via: `ucp-api-v1`,
+    },
+  };
+
+  try {
+    const client: Client = await createClient(clientToken, clientBody);
+    res.json(client);
+  } catch (error) {
+    if (
+      typeof error === "string" &&
+      error.toUpperCase() === "USER ALREADY HAS A CLIENT"
+    ) {
+      res.status(400).json({ message: "User already has a client" });
+    } else if (
+      error instanceof Error &&
+      error.message.toUpperCase().includes("TOO MANY REQUESTS")
+    ) {
+      res.status(429).json({ message: "Rate limit exceeded" });
+    } else {
+      res.status(500).json({ message: "Unable to create client" });
+    }
+  }
+};
+
+export const clientsGetV1 = async (req: Request, res: Response) => {
+  const clientToken = getClientTokenFromRequest(req);
+
+  try {
+    const client: Client = await getClient(clientToken);
+    res.json(client);
+  } catch (error) {
+    if (
+      typeof error === "string" &&
+      error.toUpperCase().includes("NOT FOUND")
+    ) {
+      res.status(404).json({ message: "Client not found" });
+    } else if (
+      error instanceof Error &&
```

Comment:
How often have you received this rate limit error?

Reply:
It happens a LOT on the cypress tests. I'm wondering if there is a way that I can improve them. The Auth0 docs say there is a limit of 10 logins per minute, and apparently sometimes my cypress tests exceed that, but only when I'm fixing/testing changes in them.